src/lib/pubkey/ec_group/ec_group.cpp

```diff
@@ -241,26 +241,38 @@ class EC_Group_Data_Map final
                }
             }
 
-         // Not found - if OID is set try looking up that way
+         /*
+         Not found in current list, so we need to create a new entry
+
+         If an OID is set, try to look up relative our static tables to detect a duplicate
+         registration under an OID
+         */
+
+         std::shared_ptr<EC_Group_Data> new_group =
+            std::make_shared<EC_Group_Data>(p, a, b, g_x, g_y, order, cofactor, oid, source);
 
          if(oid.has_value())
             {
-            // Not located in existing store - try hardcoded data set
             std::shared_ptr<EC_Group_Data> data = EC_Group::EC_group_info(oid);
-
-            if(data)
+            if(data != nullptr && !new_group->match(*data))
```

Comment:
`new_group->match_param(*data)` would be more explicit (and still short). Also, the function should take a `const` reference.

Reply:
Not understanding this - we still don't want to deref a nullptr, right? Agree re the arg being `const` ref.

Reply:
My suggestion was to rename function `EC_Group_Data::match` to `EC_Group_Data::match_param`. This would indicate to the reader that a match **by domain parameters** is attempted, which isn't obvious from the function signature alone. I didn't mean the `nullptr` check should be removed. Sorry for the ambiguous phrasing.